src/components/CategoryList.tsx

```diff
@@ -38,24 +38,66 @@ const Item = styled.li`
 const Img = styled.img`
   width: 50px;
   height: 50px;
+  border-radius: 15px;
+  margin-bottom: 3px;
 `;
 
 const Name = styled.div`
   font-size: 12px;
 `;
 
-function CategoryList() {
+type Theme = {
+  themeId: number;
+  name: string;
+  image: string;
+};
+
+function CategoryList({ onHide }: { onHide?: () => void }) {
+  const [themes, setThemes] = useState<Theme[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState<string | null>(null);
+
+  useEffect(() => {
+    setLoading(true);
+    setError(null);
+    fetch(`${import.meta.env.VITE_API_URL}/api/themes`)
+      .then((res) => {
+        if (!res.ok) throw new Error('서버 에러');
+        return res.json();
+      })
+      .then((data) => {
+        if (Array.isArray(data.data)) {
+          setThemes(data.data);
+          if (data.data.length === 0 && onHide) onHide();
+        } else {
+          setThemes([]);
+          if (onHide) onHide();
+        }
+        setLoading(false);
+      })
+      .catch(() => {
+        setError('데이터를 불러오지 못했습니다.');
+        setLoading(false);
+        if (onHide) onHide();
+      });
+  }, []);
   return (
     <Box>
       <Title>선물 테마</Title>
-      <List>
-        {categories.map((cat) => (
-          <Item key={cat.themeId}>
-            <Img src={cat.image} alt={cat.name} />
-            <Name>{cat.name}</Name>
-          </Item>
-        ))}
-      </List>
+      {loading ? (
+        <div>로딩 중 ... </div>
+      ) : error ? (
+        <div>{error}</div>
+      ) : themes.length === 0 ? null : (
+        <List>
+          {themes.map((cat) => (
+            <Item key={cat.themeId}>
+              <Img src={cat.image} alt={cat.name} />
+              <Name>{cat.name}</Name>
+            </Item>
+          ))}
+        </List>
+      )}
```

Comment:
사람마다 보기에 따라 다르지만,,,

이렇게 삼항 연산자가 겹칠때는 즉시실행함수 + early return 을 조합해주면 더 깔끔하게 작성할 수 있어요!
```suggestion
      {(() => {
        if(loading){
          return <div>로딩 중 ... </div>
        }
        if(error){
          return <div>{error}</div>
        }
        if(themes.length === 0){
          return null
        }
        return (
        <List>
          {themes.map((cat) => (
            <Item key={cat.themeId}>
              <Img src={cat.image} alt={cat.name} />
              <Name>{cat.name}</Name>
            </Item>
          ))}
        </List>
        )
      })()}
```